plugins/datastreamer_plugin/fastdds/Handler.hpp

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// This file is part of eProsima Fast DDS Monitor.
+//
+// eProsima Fast DDS Monitor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// eProsima Fast DDS Monitor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Monitor. If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * @file Handler.hpp
+ */
+
+#ifndef _EPROSIMA_PLOTJUGGLERFASTDDSPLUGIN_PLUGINS_DATASTREAMERPLUGIN_FASTDDS_HANDLER_HPP_
+#define _EPROSIMA_PLOTJUGGLERFASTDDSPLUGIN_PLUGINS_DATASTREAMERPLUGIN_FASTDDS_HANDLER_HPP_
+
+#include <memory>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/domain/DomainParticipantListener.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+
+#include "FastDdsListener.hpp"
+#include "Participant.hpp"
+
+namespace eprosima {
+namespace plotjuggler {
+namespace fastdds {
+
+/**
+ * @brief This class handles every Fast DDS entity required.
+ *
+ * It create, manage and destroy every Fast DDS entity that the process requires to instantiate.
+ * The discovery and user data received is transmitted through a FastDdsListener object.
+ *
+ * FUTURE WORK:
+ * Use a specific thread to call callbacks instead of using Fast DDS thread
+ */
+class Handler
+{
+public:
+
+    ////////////////////////////////////////////////////
+    // CREATION & DESTRUCTION
+    ////////////////////////////////////////////////////
+
+    Handler(
+            FastDdsListener* listener);
+
+    virtual ~Handler();
+
+
+    ////////////////////////////////////////////////////
+    // INTERACTION METHODS
+    ////////////////////////////////////////////////////
+
+    void connect_to_domain(
+            const uint32_t domain);
+
+    void register_type_from_xml(
+            const std::string& xml_path);
+
+    void create_subscription(
+            const std::string& topic_name);
+
+    std::shared_ptr<TopicDataBase> get_topic_data_base() const;
```

Comment:
In general for every discussion related with `TopicDataBase` ownership.
`TopicDataBase` is created by `Handler` and is shared in `Participant` (I agree that this could be a raw, as Participant is managed by Handler) but it is also shared with `DialogTopic`, an object that the Handler does not manage, and cannot verify it will not try to access TopicDataBase after Handler destruction.
This could probably be fixed with a more detailed design, but as I was in a hurry I did it this way. And I think I prefer to leave it this way for now.

Reply:
Ok let's leave it like this and maybe revisit it in the future